File: backend/SBL project/SBL.Data/Models/Domain/Category.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SBL.Data.Models.Domain
{
    public class Category
    {
        [JsonProperty("categoryId")]
        public int? CategoryId { get; set; }

        [JsonProperty("categoryName")]
        public string CategoryName { get; set; }
    }
}

File: UI/skyBookLibrary/src/app/data/models/category.ts

export class Category implements ICategory {
    public categoryId?: number
    public categoryName: string;

    static create(category: ICategory){
        if(!category) return category
        let c = new Category()
        c.categoryId = category.categoryId
        c.categoryName = category.categoryName
        return c
    }
    
}



export interface ICategory {
    categoryId?: number,
    categoryName: string
}